index.html

```diff
@@ -163,12 +239,35 @@
       .attr('r', 5);
 
     points
-      .attr('cx', function (d) { return x(d.unemployment); })
-      .attr('cy', function (d) { return y(d.inflation); });
+      .attr('cx', function (d) { return x(d.year); })
+      .attr('cy', function (d) { return y(d.value); });
 
     points.exit().remove();
 
     // HISTOGRAMS
+    var hist = d3.layout.histogram()
+       .value(function (d) { return d.year; })
+       .bins(x.ticks(binCount)
+       (txData));
+
+    var y = d3.scale.linear()
+        .domain([0, d3.max(hist, function (d) { return d.length; })])
+        .range([0, height]);
+
```

Comment:
Note you have already defined y as the linear scale for your scatterplot above. You should definitely use a different name for a different scale - this could be causing your problem.